src/ape_accounts/accounts.py

```diff
@@ -257,3 +259,39 @@ def __decrypt_keyfile(self, passphrase: str) -> HexBytes:
             return EthAccount.decrypt(self.keyfile, passphrase)
         except ValueError as err:
             raise InvalidPasswordError() from err
+
+
+def generate_account(
+    alias: str, passphrase: str, hd_path: str = ETHEREUM_DEFAULT_PATH, word_count: int = 12
+) -> Tuple[AccountAPI, str]:
+    """
+    Generate a new account.
+
+    Args:
+        extra_entropy (str): Extra entropy to use when generating the account.
+
+    Returns:
+        Tuple of AccountAPI and mnemonic for the generated account.
+    """
+    EthAccount.enable_unaudited_hdwallet_features()
+    accounts = AccountManager()
+
+    if alias in accounts.aliases:
+        raise AliasAlreadyInUseError(alias)
+    elif not isinstance(alias, str):
+        raise AccountsError(f"Alias must be a str, not '{type(alias)}'.")
+    elif is_hex(alias) and len(alias) >= 42:
+        # Prevents private keys from accidentally being stored in plaintext
+        # Ref: https://github.com/ApeWorX/ape/issues/1525
+        raise AccountsError("Longer aliases cannot be hex strings.")
+
+    if not passphrase or not isinstance(passphrase, str):
+        raise AccountsError("Account file encryption passphrase must be provided.")
+
+    account, mnemonic = EthAccount.create_with_mnemonic(num_words=word_count, account_path=hd_path)
+
+    # Write the encrypted account file
+    path = accounts.containers["accounts"].data_folder.joinpath(f"{alias}.json")
```

Comment:
We should have a method on the container that does this, is there not? If so, can we add it to simplify and share code?

Reply:
By "this" do you mean gets the file path for an alias?  Also, I don't really know what "containers" are so I feel like I'm missing some context.

Reply:
i actually had meant to delete this comment before submitting my review.
im going to refactor this core plugin later and tighten up the cli parts and container parts.